### DISCUSSION TITLE
Json file not found after executable build

### DISCUSSION THREAD

**Discussion by pablofueros** [pyinstaller/pyinstaller#8897] [Help]
Hi everyone.

I'm having a lot of trouble trying to make my app dist work. The app runs fine in local, and the building does not show any errors.

Despite all of that, once I open the executable, I get this error (if read similar stuff on #8879):

![image](https://github.com/user-attachments/assets/cb213274-c431-449c-afca-3540c4a62a10)

The command im launching is: `pyinstaller --onedir --noconsole main.py`

Note that I'm launching it using `--onedir` with debugging intentions, but I would like it to work in `--onefile` mode

My project structure is:

```
C:\USERS\PABLO\GITREPOS\SIHOT-REPORTS-COPY
│   .gitignore
│   .python-version
│   main.py
│   pyproject.toml
│   README.md
│   sihot.ico
│   uv.lock
│
└───src
    └───sihot_reports
        │   exporter.py
        │   fetcher.py
        │   loader.py
        │   model.py
        │   parser.py
        │   service.py
        │   __init__.py
        │
        └───templates
            ├───json
            │       hotels.json
            │
            └───xml
                    authentication.xml
                    invoice_get.xml
                    invoice_search.xml
                    reservation_get.xml
                    reservation_search.xml
```

The json is needed to start the program, so its the 1st error I'm getting, but the xml are also needed when requesting data and if I hardcode the json data and build the app, I get this new error when starting the app:

![image](https://github.com/user-attachments/assets/7e970fe0-cd13-4897-bddc-ea6309c588fe)

I have tried the following commands to add additional data but but they also do not work:
* `pyinstaller --onedir --noconsole --add-data="./src/sihot_reports/templates/json/hotels.json;." main.py`
* `pyinstaller --onedir --noconsole --add-data="./src/sihot_reports/templates;templates" main.py`
* `pyinstaller --onedir --noconsole --add-data="./src/sihot_reports;sihot_reports" main.py`

I would greatly appreciate your help, thank you in advance!

**Comment by rokm** [comment]
> pyinstaller --onedir --noconsole --add-data="./src/sihot_reports/templates/json/hotels.json;." main.py

This won't work, because you are collecting that particular json into top-level application directory (i.e., into the `_internal` directory itself in onedir case).

> pyinstaller --onedir --noconsole --add-data="./src/sihot_reports;sihot_reports" main.py

This should do the trick - you can check that `_internal` directory contains `sihot_reports` directory with `templates` sub-directory. As a side effect, though, it will also collect .py files from `sihot_reports` itself, which is redundant.

(But since you say that it doesn't work - what was the error message in this case?)

So 

`pyinstaller --onedir --noconsole --add-data="./src/sihot_reports/templates;sihot_reports/templates" main.py`

should collect only `sihot_reports/templates` and place it under `_internal/sihot_reports/templates`. (Note that the destination part of argument is different from what you tried to use - it is `sihot_reports/templates`, not just `templates`).

If this one doesn't work either, build with added `--clean`, and then post the error message for that build.

**Reply by pablofueros** [reply]
It worked! Thank you so much!

I also tried it using `--onefile` and works too, so I will close the discussion.

Thanks again, take care!
